designs/design_week2.md

```diff
@@ -0,0 +1,104 @@
+# CSE 231 Week 7 Project Milestone
+
+## I. What We Got Working
+
+For this week, we implemented the bare-bone functionalities of Python File-I/O, with some compromises to accomodate the current state of the compiler (no `string` class, no `list` class, etc.). In the following section, we will give a summary of the changes we've made.
+
+### 1. The Framework
+In our implementation, bascially most of the I/O functionality are implemented in Javascript. we implement some of the key functionalities as functions in JS, and import those functions to be called in the WASM code. Beneath the JS layer, the main file I/O is handled by a browser file system.
+
+In Python, file I/O is mainly interfaced through a built-in `File` class. To keep the design modular, we choose to implement the `File` class in *Python*. The class difinition code is automatically visible in the *user-code* text area every time the website is reloaded. Additionally, an `open()` global function is also defined. This allow us the leverage the implemented part of the compiler and gives us much more flexibility.
```

Comment:
I think that makes for an odd user interaction. What happens if the user deletes it? Or defines a different `File` class?

Reply:
We could append the File class definition before compilation starts, this way the user cannot accidentally delete it.

As we mentioned in the readme, we believe this design choice makes the most sense. For the conflicting definitions of the File class, we can only rely on type checking to catch the error.

Reply:
I think it's not a good idea to append code in general (if _all_ the groups do that, how long will our programs be!)

Could you have it in .wat and import it using WASM imports?